File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03802/p03802_num1_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<int> x(N), y(N);
    for (int i = 0; i < N; i++) {
        cin >> x[i] >> y[i];
    }
    
    int max_distance = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            int distance = max(abs(x[i] - x[j]), abs(y[i] - y[j]));
            max_distance = max(max_distance, distance);
        }
    }
    
    cout << max_distance << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03802/p03802_num1_parsed.py
def calculate_max_distance(N, positions):
    positions.sort()
    max_distance = 0
    for i in range(N - 1):
        distance = abs(positions[i] - positions[i + 1])
        if distance > max_distance:
            max_distance = distance
    return max_distance

N = int(input())
positions = []
for _ in range(N):
    x, y = map(int, input().split())
    positions.append(x)
    positions.append(y)

print(calculate_max_distance(N, positions))